### PR TITLE
Disable parse-time name lookup in swift-ide-test

### PR DESCRIPTION
**Pull request by slavapestov**
I have some leftover patches from my work on ASTScope. A couple of places that create an ASTContext without a CompilerInvocation still had parser lookup enabled; swift-ide-test is one of them.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: lib/Parse/ParseGeneric.cpp [hunk 1]
```diff
@@ -396,15 +395,14 @@ parseFreestandingGenericWhereClause(GenericContext *genCtx) {
...
+  auto result = parseGenericWhereClause(WhereLoc, EndLoc, Requirements);
 
   genCtx->setTrailingWhereClause(
-      TrailingWhereClause::create(Context, WhereLoc, Requirements));
+      TrailingWhereClause::create(Context, WhereLoc, EndLoc, Requirements));
+
+  if (result.isErrorOrHasCompletion() || Requirements.empty())
+    return result;
 
   return ParserStatus();
```
Comment (rintaro) [inline lib/Parse/ParseGeneric.cpp:407]: I thunk we can just return `result` here

**Review by rintaro — APPROVED** [review]
Looks great. Thank you!

File: include/swift/AST/GenericParamList.h [hunk 2]
```diff
@@ -321,7 +321,7 @@ class GenericParamList final :
 
   /// Retrieve the source range covering the where clause.
   SourceRange getWhereClauseSourceRange() const {
-    if (WhereLoc.isInvalid())
+    if (WhereLoc.isInvalid() || Requirements.empty())
       return SourceRange();
```
Comment (rintaro) [inline include/swift/AST/GenericParamList.h:325]: How about
```cpp
if (WhereLoc.isInvalid())
  return SourceRange();
if (Requirements.empty())
  return WhereLoc;
```

**Comment by slavapestov** [comment]
Thanks for the review, @rintaro!

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**